### PR TITLE
FIX: Bookmark auto delete preference usage and default value

### PR DESCRIPTION
**Pull request by martin-brennan**
This commit fixes an issue where the chat message bookmarks
did not respect the user's `bookmark_auto_delete_preference`
which they select in their user preference page.

Also, it changes the default for that value to "keep bookmark and clear reminder"
rather than "never", which ends up leaving a lot of expired bookmark
reminders around which are a pain to clean up.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/bookmark.js [hunk 1]
```diff
@@ -60,7 +60,9 @@ export default Component.extend({
       userTimezone: this.currentUser.user_option.timezone,
       showOptions: false,
       _itsatrap: new ItsATrap(),
-      autoDeletePreference: this.model.autoDeletePreference || 0,
+      autoDeletePreference:
+        this.model.autoDeletePreference ||
+        AUTO_DELETE_PREFERENCES.CLEAR_REMINDER,
```
Comment (nbianca) [inline app/assets/javascripts/discourse/app/components/bookmark.js:65]: 👍🏻

File: lib/bookmark_manager.rb [hunk 2]
```diff
@@ -136,20 +138,30 @@ def update_topic_user_bookmarked(topic, opts = {})
...
 
   def bookmark_model_options_with_defaults(options)
-    model_options = {
-      pinned: options[:pinned]
-    }
+    model_options = { pinned: options[:pinned] }
 
     if options[:auto_delete_preference].blank?
-      model_options[:auto_delete_preference] = Bookmark.auto_delete_preferences[:never]
+      model_options[:auto_delete_preference] = if user_auto_delete_preference.present?
```
Comment (nbianca) [inline lib/bookmark_manager.rb:152]: Not sure if my personal preference, but I think this is a bit more clear:

```ruby
model_options[:auto_delete_preference] = 
  options[:auto_delete_preference].presence ||
  @guardian.user.user_option&.bookmark_auto_delete_preference.presence ||
  Bookmark.auto_delete_preferences[:clear_reminder]
```

**Review by nbianca — APPROVED** [review]
Are we going to announce somewhere more visible the change in the default option from "never" to "keep bookmark and clear reminder"?

File: lib/bookmark_manager.rb (hunk 2 above)
Comment (martin-brennan) [reply lib/bookmark_manager.rb:152]: I don't find that at all more readable personally :) Anyway, I've run this through `syntax_tree` and this is how it formatted it, so leaving as is

**Comment by martin-brennan** [comment]
> Are we going to announce somewhere more visible the change in the default option from "never" to "keep bookmark and clear reminder"?

Yes will ann

**Pull request merged**